src/components/Auth/PasswordInput.vue

```diff
@@ -0,0 +1,84 @@
+<script>
+import { defineComponent } from 'vue';
+
+import BaseTextInput from '../global/BaseTextInput.vue';
+import ViewOffIcon from '@/assets/icons/view-off-icon.svg';
+import ViewOnIcon from '@/assets/icons/view-on-icon.svg';
+
+export default defineComponent({
+  name: 'PasswordInput',
+  components: {
+    BaseTextInput,
+    ViewOnIcon,
+    ViewOffIcon
+  },
+  props: {
+    modelValue: {
+      type: String
+    }
+  },
+  data() {
+    return {
+      isPasswordVisible: false,
+      inputType: 'password',
+      inputValue: ''
```

Comment:
```suggestion
    inputValue() {
      this.$emit('update:modelValue', this.inputValue);
      this.$emit('focus');
    }
```
Actually we... do not need this 😄
In order to have this "nested" v-model where we use v-model on a component which underneath want to do just that, we can do so by instead of assigning just the v-model, assign manually both the value binding and event handler.
So assuming we already have our `modelValue` prop:

```vue
  <BaseTextInput
    :model-value="modelValue"
    @update:modelValue="$emit('update:modelValue', $event)"
  />
```

This way we pass the value deeper and every event we re-emit upwards and that's it! 😄

Reply:
I tried to do it the way You suggest initially and was struggling so, a bit frustrated, I went with the local data handling..
Now I know what was wrong and got the idea! :boom: